tests/test_calculate.py

```diff
@@ -0,0 +1,206 @@
+
+    def _mock_method_and_keywords(method_name):
+        return 'mock_method', 'mock_keywords'
+
+    monkeypatch.setattr(
+        'mlptrain.configurations.calculate._method_and_keywords',
+        _mock_method_and_keywords,
+    )
+
+
+@pytest.fixture
+def mock_autode(monkeypatch):
+    """Fixture to mock autode Species and Calculation"""
+    monkeypatch.setattr('autode.species.Species', MockSpecies)
+    monkeypatch.setattr('autode.calculation.Calculation', MockCalculation)
+
+
+@pytest.fixture
+def set_config(monkeypatch):
+    """Fixture to set required config values for ORCA and Gaussian keywords"""
+    Config.orca_keywords = ['PBE', 'def2-SVP', 'EnGrad']
+    Config.gaussian_keywords = ['B3LYP', '6-31G(d)', 'Force']
+
+
+# Tests
+
+
+def test_run_autode_success(
+    mock_autode, mock_method_and_keywords, configuration
+):
+    """Test run_autode for a successful calculation"""
+
+    run_autode(configuration, method_name='mock_method', n_cores=1)
+
+    # Assertions to verify configuration attributes were set as expected
+    assert (configuration.forces.true == np.array([0.1, -0.2, 0.3])).all()
+    assert configuration.energy.true == -1.0
+    assert (configuration.partial_charges == [0.5, -0.5]).all()
+
+
+def test_run_autode_failed_energy(
+    mock_autode, mock_method_and_keywords, configuration, capsys
+):
+    """Test run_autode when energy calculation fails but gradients succeed"""
+
+    # Set up mock Calculation to simulate energy calculation failure
+    calc_instance = MockCalculation(
+        name='tmp', molecule=None, method=None, keywords=None, n_cores=1
+    )
+    calc_instance._energy_success = False  # Fail energy calculation
+    calc_instance._gradient_success = True  # Succeed in gradient calculation
+
+    with pytest.MonkeyPatch.context() as mp:
+        mp.setattr(
+            'autode.calculation.Calculation',
+            lambda *args, **kwargs: calc_instance,
+        )
+        run_autode(configuration, method_name='mock_method', n_cores=1)
+
+    captured = capsys.readouterr()
+    assert 'Failed to calculate the energy' in captured.out
+    assert configuration.energy.true is None
+
+
+def test_method_and_keywords_success():
+    """Test _method_and_keywords for valid methods"""
+    methods = {'orca': 'orca', 'g09': 'g09', 'g16': 'g16', 'xtb': 'xtb'}
+    for method_name, expected in methods.items():
+        method, keywords = _method_and_keywords(method_name)
+        assert (
+            method.name == expected
+        )  # Mocked ORCA, G09, etc., should have these names
+
+
+def test_method_and_keywords_invalid():
+    """Test _method_and_keywords raises ValueError for an invalid method"""
+    with pytest.raises(ValueError, match='Unknown method'):
+        _method_and_keywords('invalid_method')
+
+
+def test_orca_keywords_success(set_config):
+    """Test _orca_keywords retrieves the ORCA keywords from Config"""
+    keywords = _orca_keywords()
+    assert keywords == Config.orca_keywords
```

Comment:
Does it still fail if you try this?

```suggestion
    assert keywords._list == Config.orca_keywords._list
```